### ISSUE TITLE
Type mismatch: cannot convert from Object to Element	AadlUtil.java	

### ISSUE THREAD

**Issue by brlarson** [osate/osate2-core#677]
missing type coercion; change 
		if (object instanceof IAdaptable) {
			theElement = ((IAdaptable) object).getAdapter(Element.class);
			if (theElement != null) {
				return theElement;
			}
		}
to
		if (object instanceof IAdaptable) {
			theElement = (Element)((IAdaptable) object).getAdapter(Element.class);
			if (theElement != null) {
				return theElement;
			}
		}

**Comment by lwrage** [comment]
In which file is that?

**Comment by brlarson** [comment]
AadlUtil.java
in org.osate.aadl2.modelsupport.util

**Comment by lwrage** [comment]
I don't think so. Method getAdapter is generic and returns whatever type is passed as a parameter. Do you get an exception somewhere?

**Comment by brlarson** [comment]
No exception, but error marker in text with message use as the title of this issue.
I don't run with errors, so needed to fix it by inserting the (Element).

**Comment by lwrage** [comment]
No error marker here. What's your Java version? What's your setting for JDK compliance level/Java source compatibility setting in Eclipse?

**Comment by brlarson** [comment]
I'd been using Luna SR2 (4.4.2) because I had some problems with Mars.
Trying Mars again, I get 100+ errors after build, so I pulled the Git develop branch afresh.
Still have 141 errors, but they're all in projects org.osate.xtext.*
Trying Check for Updates; selected just Eclipse Modeling Tools and the Xtext updates.
Upon restart, have "Mars.2".  Build-Clean-Build still has 141 errors:


Most, but not all errors report as "Xtend Problem".
Trying to resolve the first error:
import org.eclipse.xtend.lib.annotations.Accessors
I added org.eclipse.xtend.core and org.eclipse.xtend.standalone to the dependencies of org.osate.xtext.aadl2.
No improvement.  But I noticed manifest errors in:
org.eclipse.xtext;bundle-version="[2.8.0,2.9.0)";visibility:=reexport, 
With Unsatisfied Version Constraint.
Updating Xtext must have exceeded the 2.9.0 limit.
It was about here, that I gave up on my previous attempt to use Mars and the latest OSATE develop branch pull, and went back to Luna SR2, added the type conversion, and kept working.
